src/eredis_client.erl

```diff
@@ -457,6 +457,12 @@ do_sync_command(Socket, _, Command) ->
             {error, Reason}
     end.
 
+close_transport(State, Socket) ->
+    case State#state.transport of
+        tls -> ssl:close(Socket);
```

Comment:
I think it's more nice to pattern-match in the function clause:

```Erlang
close_transport(#state{transport = tls}, Socket) -> ssl:close(Socket);
close_transport(#state{transport = tcp}, Socket) -> tcp:close(Socket).
```

Btw, perhaps "close_socket" is a better name.